src/command/streams/WriteEventsToStream.ts

```diff
@@ -154,13 +154,25 @@ export class WriteEventsToStream extends Command {
             message.getMetadataMap().set("content-type", "application/json");
             const data = JSON.stringify(event.payload.payload);
             message.setData(Buffer.from(data, "binary").toString("base64"));
+            if (event.metadata) {
+              const metadata = JSON.stringify(event.metadata);
+              message.setCustomMetadata(
+                Buffer.from(metadata, "binary").toString("base64")
+              );
+            }
             break;
           }
           case "binary": {
             message
               .getMetadataMap()
               .set("content-type", "application/octet-stream");
             message.setData(event.payload.payload);
+            if (event.metadata) {
+              const metadata = JSON.stringify(event.metadata);
+              message.setCustomMetadata(
+                Buffer.from(metadata, "binary").toString("base64")
+              );
+            }
```

Comment:
Sure - didn't want to over invest till I got general acceptance.

Reply:
@George-Payne how do you run the tests locally? Its a PITA to push up to github to debug

Reply:
Sorry, the default docker image is currently private, I need to change that.

You can set it to the public one like this:

```
yarn cross-env EVENTSTORE_IMAGE=eventstore/eventstore:latest jest
```

(and you need docker and docker-compose installed)

Reply:
(and you can run a specific test with: 

```
yarn cross-env EVENTSTORE_IMAGE=eventstore/eventstore:latest jest writeEventsToStream
```
)

Reply:
FWIW - jest fails by choking on: `const version = process.env.EVENTSTORE_IMAGE ?? "github:ci";`

Using node v12.18.4 and everything else derived from `package.json`

Reply:
using node v14.01 helped but the tests fail miserably as it tries to run before the store is ready and available.

Is there a way to skip the startup and just use the default dev ports and dev (no) security? Otherwise it would be useful to have a clearer recipe to make it easier to provide PRs. I'm not frustrated, this is just a gentle nudge.

Reply:
> Using node v12.18.4

Thats an oversight on my part, I switched the tests typescript target to ES2020 to so I could use bigint literals, forgot that that would break LTS node. I've opened an issue (#43) and started a PR to fix it. (#44)

Reply:
> the tests fail miserably as it tries to run before the store is ready and available.

That definately shouldn't happen. I think that the issue is that it's failing to pull the image within the specified timeout. Jest has the unfortunate behavior of barreling on with the tests even if the startup times out or throws, and then all the tests fail with being unable to connect. 

Can you try running `docker pull eventstore/eventstore:latest` before starting the tests?

Reply:
> Is there a way to skip the startup and just use the default dev ports and dev (no) security?

Not without modifying the test, no.

> Otherwise it would be useful to have a clearer recipe to make it easier to provide PRs. 

In theory the tests should run with just a `yarn test`, but obviously this hasn't happened in practice. This should mostly be remedied by setting the default docker image to the public image (#42), but maybe I could add a prefilght check that checks if you have docker / docker-compose and the image already pulled, and logs a big warning if not. 

The testing part of the readme does mention needing docker / docker-compose, and how to select an image to run against: https://github.com/EventStore/EventStore-Client-NodeJS#run-tests but thats not very helpful if the tests dont give an idication that that is the issue.

Reply:
> > the tests fail miserably as it tries to run before the store is ready and available.
>
> That definately shouldn't happen. [...]

And just like that, the CI fails with the store not being ready:  #45 

Is this the same issue you saw?

Reply:
Well I do have docker-compose installed and I've had the images already loaded to no avail. Pushing things up to github to debug, while a PITA, is no big deal, so I'll leave it there. 

If you would like to dig deeper into my problems, I'd be happy to help, otherwise the team at my company would like to start replacing our use of node-eventstore-client with this. We need metadata, so we'd like to get this PR approved.